src/main/java/mate/academy/spring/controller/AuthenticationController.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.impl.response.UserResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.request.UserRegistrationDto;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.security.AuthenticationService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    private final UserResponseMapper userResponseMapper;
```

Comment:
```suggestion
    private final DtoResponseMapper<UserResponseDto, User> userDtoResponseMapper;
```
You must depend from abstraction, not realization, see check_list